### ISSUE TITLE
null pointer risk in OneToManyProcessor.cpp

### ISSUE THREAD

**Issue by yuanchao0310** [lynckia/licode#1372]
**Description**
application call OneToManyProcessor.cpp close() function, close() call closeAll(),  and closeAll() will set feedbackSink_=nullptr;

in OneToManyProcessor::deliverFeedback_ function, it will use feedbackSink_. this deliverFeedback_ is called in MediaStream worker thread. The MediaStream worker thread may different with application thread. so there is a risk that can cause crash.

I think one solution:
in deliverFeedback_() function lock monitor_mutex_;
in closeAll() function, put feedbackSink_=nullptr after boost::unique_lock<boost::mutex> lock(monitor_mutex_);

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Licode commit/release where the issue is happening**

**Additional environment details (Local, AWS, Docker, etc.):**

**Comment by jcague** [comment]
yes, we should finish up the work we started to adopt smart pointers and also fix this case with weak_ptr I think. We'll work on it asap.
